crates/rpc/rpc/src/eth/signer.rs

```diff
@@ -26,14 +32,35 @@ pub(crate) trait EthSigner: Send + Sync {
         request: TypedTransactionRequest,
         address: &Address,
     ) -> Result<TransactionSigned>;
+
+    /// Encodes and signs the typed data according EIP-712. Payload must implement Eip712 trait.
+    fn sign_typed_data(&self, address: Address, payload: &TypedData) -> Result<Signature>;
 }
 
 /// Holds developer keys
 pub(crate) struct DevSigner {
     addresses: Vec<Address>,
     accounts: HashMap<Address, SecretKey>,
+    sign_engine: Secp256k1<All>,
 }
 
+impl DevSigner {
+    fn get_key(&self, account: Address) -> Result<&SecretKey> {
+        self.accounts.get(&account).ok_or(SignError::NoAccount)
+    }
+    fn sign_hash(&self, hash: H256, account: Address) -> Result<Signature> {
+        let secret = self.get_key(account)?;
+        let message = &Message::from_slice(hash.as_bytes()).map_err(|_| SignError::CouldNotSign)?;
+        let (rec_id, data) =
+            self.sign_engine.sign_ecdsa_recoverable(message, secret).serialize_compact();
+        let signature = Signature {
+            r: U256::try_from_be_slice(&data[..32]).ok_or(SignError::CouldNotSign)?,
+            s: U256::try_from_be_slice(&data[32..64]).ok_or(SignError::CouldNotSign)?,
+            odd_y_parity: rec_id.to_i32() != 0,
+        };
+        Ok(signature)
```

Comment:
looks like we already have this function:

https://github.com/paradigmxyz/reth/blob/main/crates/interfaces/src/test_utils/generators.rs#L77-L78

we should move this to primitives imo, so we can reuse this

Reply:
Done 👍